java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java

```diff
@@ -118,12 +132,14 @@ public String toString() {
   }
 
   private static AnalyzerMessage.TextSpan textSpanBetween(SyntaxToken firstSyntaxToken, SyntaxToken lastSyntaxToken) {
-    return new AnalyzerMessage.TextSpan(
+    AnalyzerMessage.TextSpan location = new AnalyzerMessage.TextSpan(
       firstSyntaxToken.line(),
       firstSyntaxToken.column(),
       lastSyntaxToken.line(),
       lastSyntaxToken.column() + lastSyntaxToken.text().length()
     );
+    Preconditions.checkState(!location.isEmpty(), "Issue location should not be empty");
```

Comment:
@benzonico : I tried different things but I was not convinced. I agree that we should be more helpful at this point. What do you think of : 
```
Preconditions.checkState(!location.isEmpty(), "Invalid issue location: Issue location is empty when trying reporting on "
      + "(" + firstSyntaxToken.line() + "," + firstSyntaxToken.column() +").");
```